nix_update/update.py

```diff
@@ -152,25 +152,38 @@ def update_cargo_deps_hash(opts: Options, filename: str, current_hash: str) -> N
     replace_hash(filename, current_hash, target_hash)
 
 
-def update_cargo_lock(opts: Options, filename: str, dst: str) -> None:
+def update_cargo_lock(opts: Options, filename: str, dst: CargoLockInSource | CargoLockInStore) -> None:
```

Comment:
I found the previous type a bit hard to read so I introduced a new more descriptive type.